tasks/resolve_reference.rb

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative '../../ruby_task_helper/files/task_helper.rb'
+require 'json'
+require 'net/http'
+
+class Vault < TaskHelper
+  class VaultHTTPError < TaskHelper::Error
+    def initialize(response)
+      err = JSON.parse(response.body)['errors']
+      m = String.new("#{response.code} \"#{response.msg}\"")
+      m << ": #{err.join(';')}" unless err.nil?
+      super(m, 'bolt.plugin/vault-http-error')
+    end
+  end
+
+  class ValidationError < TaskHelper::Error
+    def initialize(m)
+      super(m, 'bolt.plugin/validation-error')
+    end
+  end
+
+  # All requests for secrets must have a token in the request header
+  TOKEN_HEADER = "X-Vault-Token"
+
+  # Default header for all requests, including auth methods
+  DEFAULT_HEADER = {
+    "Content-Type" => "application/json",
+    "Accept" => "application/json"
+  }.freeze
+
+  ENV_OPTS = {
+    server_url: ENV['VAULT_ADDR'],
+    auth: ENV['VAULT_TOKEN'],
+    cacert: ENV['VAULT_CACERT']
+  }.freeze
+
+  def validate_options(opts)
+    %i[server_url auth path].each do |key|
+      unless opts[key]
+        raise ValidationError, "Vault plugin requires #{key} to be configured"
+      end
+    end
+  end
+
+  def task(opts)
+    # Precedence: Inventory overrides config overrides env
+    merged = ENV_OPTS.merge(opts)
+
+    validate_options(merged)
+    header = {
+      TOKEN_HEADER => request_token(merged[:auth], merged)
+    }
+
+    response = request(:Get, uri(merged), merged, nil, header)
+
+    { "value" => parse_response(response, merged) }
+  end
+
+  # Request uri - built up from Vault server url and secrets path
+  def uri(opts, path = nil)
+    url = opts[:server_url]
+
+    # Handle the different versions of the API
+    if opts[:version] == 2
+      mount, store = opts[:path].split('/', 2)
+      opts[:path] = [mount, 'data', store].join('/')
+    end
+
+    path ||= opts[:path]
+
+    URI.parse(File.join(url, "v1", path))
+  end
+
+  # Configure the http/s client
+  def client(uri, opts)
+    client = Net::HTTP.new(uri.host, uri.port)
+
+    if uri.scheme == 'https'
+      cacert = opts[:cacert]
+      unless cacert
+        raise ValidationError, "Vault plugin requires cacert to be configured when connecting over https"
+      end
+      client.use_ssl = true
+      client.ssl_version = :TLSv1_2
+      client.ca_file = cacert
+      client.verify_mode = OpenSSL::SSL::VERIFY_PEER
```

Comment:
You will need to `require 'openssl'` to use this constant.

Reply:
I am unable to get a test env set up with tls. I'm not sure how important that is to verify. Instead of continuing to struggle with it i'm going to ask if anybody else has had success getting that set up.